homeassistant/components/fritz/__init__.py

```diff
@@ -60,6 +64,26 @@ def _async_unload(event):
     return True
 
 
+@callback
+def async_device_setup(
+    hass: HomeAssistant, entry: ConfigEntry, fritz_tools: FritzBoxTools
+):
+    """Set up a device that is online."""
+    dev_reg = device_registry.async_get(hass)
+    entry = dev_reg.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        name=entry.title,
+        connections={(device_registry.CONNECTION_NETWORK_MAC, fritz_tools.mac)},
+        # This is duplicate but otherwise via_device can't work
+        identifiers={(DOMAIN, fritz_tools.mac)},
+        manufacturer=fritz_tools.device_info["manufacturer"],
+        model=fritz_tools.device_info["model"],
+        sw_version=fritz_tools.device_info["sw_version"],
+    )
+
+    return True
```

Comment:
Nothing is checking this.  We can remove it

Reply:
Done